test/pubspec_test.dart

```diff
@@ -592,5 +592,61 @@ executables:
         expect(pubspec.executables['command'], equals('command'));
       });
     });
+
+    group("web", () {
+      test("can be empty", () {
+        var pubspec = new Pubspec.parse('web: {}', sources);
+        expect(pubspec.webCompiler, isEmpty);
+      });
```

Comment:
I believe in general we allow users to explicitly specify `null` for fields that are allowed to not exist, which means `web:` without anything after it is valid.